lib/logflare/users/users.ex

```diff
@@ -186,10 +186,15 @@ defmodule Logflare.Users do
   def get_related_user_id(map) do
     case map do
```

Comment:
This is a separate conversation but I wonder if something could be done to remove the number of "sources" here.

Also, it is worth noting that you are caching whole structs, but you may only need the `user_id` field. For example, `Cache.get_endpoint_query` seems to be used exclusively here and you don 't care about the other fields, so there is no need to store or read them. You may benefit from adding a `get_user_id_by_endpoint_query`. This may also apply to other operations, but please double check!